### PR TITLE
Update the status of DevOps roadmap for upgrade section and the 3.0 section

### PR DESCRIPTION
**Pull request by LinuxSuRen**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Mentioned as the title.

**Which issue(s) this PR fixes**:
None.

**Special notes for reviewers**:
None.

**Additional documentation, usage docs, etc.**:
None.

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -124,11 +126,7 @@ KubeSphere Roadmap demonstrates a list of open source product development plans
 ### DevOps:
 
 - [x] Upgrade Jenkins Version to 2.249.1. [#2618](https://github.com/kubesphere/kubesphere/issues/2618)
-- [x] Using Jenkins distribution solution to deploy Jenkins, [#2182](https://github.com/kubesphere/kubesphere/issues/2182)
-- [x] Using human-readable error message for pipeline cron text , [#2919](https://github.com/kubesphere/kubesphere/issues/2919)
-- [ ] Using human-readable error message for S2I, [#140](https://github.com/kubesphere/s2ioperator/issues/140)
-
-- [ ] https://github.com/kubesphere/tower/issues/31)
```
Comment (LinuxSuRen) [inline docs/roadmap.md:131]: hi @yuswift , this record should be here. I don't know where this record should be.

see also https://github.com/kubesphere/kubesphere/pull/3412/files#diff-6a17656d764b8e1930354f341cf7f47960b63845c3046edcd8bbf282aaa2f464R130

File: docs/roadmap.md [hunk 2]
```diff
@@ -171,12 +169,12 @@ KubeSphere Roadmap demonstrates a list of open source product development plans
...
-- [ ] Image registry optimization.
-- [ ] Pipeline support integration with JIRA.
-- [ ] Pipeline integrates the notification of KubeSphere.
-- [ ] Pipeline integrates KubeSphere custom monitoring.
+- [x] Create / Edit Pipeline Process Optimization.
+- [x] S2I/B2I supports webhook.
+- [x] Image registry optimization.
+- [x] Pipeline support integration with JIRA.
+- [x] Pipeline integrates the notification of KubeSphere.
+- [x] Pipeline integrates KubeSphere custom monitoring.
```
Comment (LinuxSuRen) [inline docs/roadmap.md:177]: This is part of roadmap 3.0. They are all finished.

**Review by calvinyv — APPROVED** [review]

**Pull request merged**